### ISSUE TITLE
DoT/DoH-Queries and responses are not logged in pcap-logging-mode

### ISSUE THREAD

**Issue by tomarcade** [dmachard/go-dns-collector#157]
Hi

Using the latest go-dns-collector (0.24.0), DoH and DoT queries AND responses are not logged in PCAP logging mode. In the stdout and logfile-mode, these queries are logged correctly.

Output from stdout and the logfile...
```
#DOH
2022-11-10T07:45:11.365021544Z pc CLIENT_QUERY NOERROR 10.100.2.47 35675 INET DOH 58b www.microsoft.com A 0.000000
2022-11-10T07:45:11.365342574Z pc CLIENT_RESPONSE NOERROR 10.100.2.47 35675 INET DOH 247b www.microsoft.com A 0.000000

#DOT
2022-11-10T07:49:14.405353046Z pc CLIENT_QUERY NOERROR 10.100.2.47 44411 INET DOT 58b www.microsoft.com A 0.000000
2022-11-10T07:49:14.410022956Z pc CLIENT_RESPONSE NOERROR 10.100.2.47 44411 INET DOT 247b www.microsoft.com A 0.000000
```

..., but the PCAP is empty for DoH and DoT-queries.

My current config:
```
multiplexer:
  collectors:
    - name: dnstap_remote
      dnstap:
        sock-path: /run/dnsdist/unix_remote.sock
        tls-support: false
      transforms:
        filtering:
          log-queries: true
          log-replies: true

  loggers:
    - name: logfile
      logfile:
       enable: true
       file-path: /var/log/dns/dns.log
       mode: text

    - name: console
      stdout:
        mode: text

    - name: pcap
      pcapfile:
        file-path: /var/log/dns/dns.cap
        max-size: 100
        max-files: 10
        compress: false

  routes:
    - from: [ dnstap_remote ]
      to: [ console, logfile, pcap ]
```

**Comment by dmachard** [comment]
It is the expected behaviour for the pcap logger regarding DoT and DoH traffic.
What do you expect in the pcap file:
- a conversion to the DNS protocol ? If we do that, we lost the information regarding the origin protocol (DoH or DoT)
- or HTTPS or TLS protocol? but without visibility on it because of cipher

**Comment by tomarcade** [comment]
Mmh..., but if you would do a conversion to the DNS protocol, you could at least see the destination port which is 443 (DoH) or 853 (DoT) and you could match the destination port to the right protocol?
But what I've tested is, that it seems difficult too to get the correct PCAP log for dns traffic, which comes with TCP instead of plain UDP. I assume that's a similar reason because of TCP (handshake, ACKs, etc.etc.).

**Comment by dmachard** [comment]
Currently, the pcap encoding is simple with plain UDP only, but it's can be improved.

Ok to set the destination port to 443 or 853 with documentations to explain that.
Regarding TCP encoding, can you open a new issue ?

**Comment by tomarcade** [comment]
Issue for TCP-Encoding: https://github.com/dmachard/go-dns-collector/issues/167#issue-1449495041

**Comment by dmachard** [comment]
implemented in the following beta release https://github.com/dmachard/go-dns-collector/releases/tag/v0.30.0-beta4

**Comment by dmachard** [comment]
Implemented in release v0.30.0.
